src/interpreter.rs

```diff
@@ -120,3 +129,1207 @@ bitflags::bitflags! {
+                        OP_1NEGATE | OP_1 | OP_2 | OP_3 | OP_4 | OP_5 | OP_6 | OP_7 | OP_8
+                        | OP_9 | OP_10 | OP_11 | OP_12 | OP_13 | OP_14 | OP_15 | OP_16 => {
+                            // ( -- value)
+                            let bn = ScriptNum::from(u8::from(pv)) - u8::from(OP_RESERVED).into();
+                            stack.push_back(bn.getvch());
+                            // The result of these opcodes should always be the minimal way to push the data
+                            // they push, so no need for a CheckMinimalPush here.
+                        }
+                        _ => {
+                            if pv <= OP_PUSHDATA4 {
+                                if require_minimal && !check_minimal_push(&vch_push_value, pv) {
+                                    return set_error(ScriptError::MinimalData);
+                                }
+                                stack.push_back(vch_push_value.clone());
+                            } else {
+                                return set_error(ScriptError::BadOpcode);
+                            }
+                        }
+                    }
+                }
+            }
+            Opcode::Operation(op) => {
+                // Note how OP_RESERVED does not count towards the opcode limit.
+                op_count += 1;
+                if op_count > 201 {
+                    return set_error(ScriptError::OpCount);
+                }
+
+                if op == OP_CAT
+                    || op == OP_SUBSTR
+                    || op == OP_LEFT
+                    || op == OP_RIGHT
+                    || op == OP_INVERT
+                    || op == OP_AND
+                    || op == OP_OR
+                    || op == OP_XOR
+                    || op == OP_2MUL
+                    || op == OP_2DIV
+                    || op == OP_MUL
+                    || op == OP_DIV
+                    || op == OP_MOD
+                    || op == OP_LSHIFT
+                    || op == OP_RSHIFT
+                    || op == OP_CODESEPARATOR
+                {
+                    return set_error(ScriptError::DisabledOpcode); // Disabled opcodes.
+                }
+
+                if exec || (OP_IF <= op && op <= OP_ENDIF) {
+                    match op {
+                        //
+                        // Control
+                        //
+                        OP_NOP => (),
+
+                        OP_CHECKLOCKTIMEVERIFY => {
+                            // This was originally OP_NOP2 but has been repurposed
+                            // for OP_CHECKLOCKTIMEVERIFY. So, we should act based
+                            // on whether or not CLTV has been activated in a soft
+                            // fork.
+                            if !flags.contains(VerificationFlags::CHECKLOCKTIMEVERIFY) {
+                                if flags.contains(VerificationFlags::DiscourageUpgradableNOPs) {
```

Comment:
`SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS`, on the other hand, is only enforced as a standard rule.